torch/utils/benchmark/utils/timer.py

```diff
@@ -142,13 +243,34 @@ def stop_hook(times) -> bool:
             task_spec=self._task_spec
         )
 
-    def blocked_autorange(self, callback=None, min_run_time=0.2):
+    def blocked_autorange(
+        self,
+        callback: Optional[Callable[[int, float], NoReturn]] = None,
+        min_run_time: float = 0.2,
+    ) -> common.Measurement:
+        """Measure many replicates while keeping timer overhead to a minimum.
+
+        This method repeats `stmt` until a time quota (set by `min_run_time`)
+        has been met. It dynamically chooses the block size by balancing two
+        criteria:
+            1) A small block size results in more replicates and generally
+               better statistics.
+            2) A large block size better amortizes the cost of `timer`
+               invocation, and results in a less biased measurement. This is
+               important because CUDA syncronization time is non-trivial
+               (order single to low double digit microseconds) and would
+               otherwise bias the measurement.
+
+        Returns:
+            A `Measurement` object which can be used to compute statistics.
+            (mean, median, etc.)
```

Comment:
that contains measured runtimes and repetition counts of the blocks? Block is still undefined, lol